node_package/src/clientStartup.ts

```diff
@@ -304,12 +304,7 @@ export function clientStartup(context: Context): void {
 
   debugTurbolinks('Adding DOMContentLoaded event to install event listeners.');
 
-  // So  long as the document is not loading, we can assume:
-  // The document has finished loading and the document has been parsed
-  // but sub-resources such as images, stylesheets and frames are still loading.
-  // If lazy asynch loading is used, such as with loadable-components, then the init
-  // function will install some handler that will properly know when to do hyrdation.
-  if (document.readyState !== 'loading') {
+  if (document.readyState === 'complete') {
```

Comment:
This is the way the code used to be, which I changed in https://github.com/shakacode/react_on_rails/commit/7b301f9e87b4f27b9197207f8e210c38095c551b.

We need to ensure that we don't break the use case of loadable-components.

Maybe we should allow an opt-in to loading  scripts so long as the `readState !== 'loading'`?

CC: @Romex91 @AbanoubGhadban